proposals/kubeflow-versioning.md

```diff
@@ -0,0 +1,54 @@
+# Kubeflow Versioning
+
+## Motivation
+
+Regularize the Kubeflow release schedule.
+
+## Proposal
+
+In brief, this proposal would switch the release cycle to this:
+
+![Kubeflow Versioning](diagrams/kubeflow-versioning.png)
+
+The changes this would introduce are:
+
+- A new version of Kubeflow would be released every 2 months
+- The deadline would be strict; if something is not ready for a particular release, it will be bumped to the next one
+- Each version would be supported (i.e. bugs/security issues fixed) for 4 months, except for for yearly Extended Support (ES) versions that would be supported for 16 months
+- Kubeflow would be versioned according to a YY.MM.N scheme, where YY and MM are the two-digit year and month of the release date, and N is the patch number. N starts at 0 for each YY.MM release, and is incremented by one for each patch release.
+
+This proposal would not change:
+
+- The release cycle of individual Kubeflow components
+
+## Motivation
+
+- Strict release schedule
+  - It is not easy to plan around the current release schedule, as it does not have hard deadlines.
+- Bumping components to subsequent releases
+  - Kubeflow components must not block releases for this proposal to work, otherwise Kubeflow releases can't be strict.
+- 2 month release cycle
```

Comment:
For this to work there might be the need to create a `rolling leadership scheme` or something that ensures there is accountable leadership and capacity in the community to do the work.